### PR TITLE
All teacher lessons

### PR DESCRIPTION
**Pull request by dborski**
## Description
Implements `GET api/v1/teachers/:teacher_id/lessons` endopint to grab all lessons (along with questions and answers for those lessons) for a specific teacher

Completes User Story #20 
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## RSpec Results
- 100%

### CODE REVIEW

File: api/tests/test_views.py [hunk 1]
```diff
@@ -155,6 +165,17 @@ def test_can_get_lesson(self):
...
+  def test_can_get_all_lessons_for_teacher(self):
+    response = self.client.get('/api/v1/teachers/%s/lessons/' % self.teacher1.id)
+
+    self.assertEqual(response.status_code, 200)
+    self.assertIsNotNone(response.data[0]['name'])
+    self.assertIsNotNone(response.data[1]['name'])
+    self.assertIsInstance(response.data[0]['questions'], list)
+    self.assertIsInstance(response.data[1]['questions'], list)
+    self.assertIsInstance(response.data[0]['questions'][0]['answers'], list)
+    self.assertIsInstance(response.data[1]['questions'][0]['answers'], list)
```
Comment (Lithnotep) [inline api/tests/test_views.py:177]: Solid ttd here

File: api/views.py [hunk 2]
```diff
@@ -43,8 +43,10 @@ class TeacherLesson(APIView):
   parser_classes = [JSONParser]
 
   def get(self, request, pk):
-    lesson = Lesson.objects.get(pk=pk)
-    return Response(LessonSerializer(lesson).data)
+    teacher = Teacher.objects.get(pk=pk)
+    lessons = teacher.lesson_set.all()
+    
+    return Response(LessonSerializer(lessons, many=True).data)
```
Comment (Lithnotep) [inline api/views.py:49]: so the lesson serializer here takes care of getting the questions and answers as well?

Comment (dborski) [reply api/views.py:49]: Ya take a look at the LessonSerialzier and you can see it also has a questions attribute, which then calls a chain of other serializers to get all the questions and answers.

File: api/tests/test_views.py [hunk 3]
```diff
@@ -82,6 +82,16 @@ def setUp(self):
...
+    self.lesson2 = self.teacher1.lesson_set.create(name='name2', description='description2')
+
+    self.question3 = self.lesson2.question_set.create(question='question1', reading='reading1')
+    self.answer5 = self.question3.answer_set.create(answer='answer1', correct=False)
+    self.answer6 = self.question3.answer_set.create(answer='answer2', correct=True)
+
+    self.question4 = self.lesson2.question_set.create(question='question2', reading='reading2')
+    self.answer7 = self.question4.answer_set.create(answer='answer3', correct=True)
+    self.answer8 = self.question4.answer_set.create(answer='answer4', correct=False)
+
```
Comment (melatran) [inline api/tests/test_views.py:94]: tests look great

**Review by melatran — APPROVED** [review]

**Review by Lithnotep — APPROVED** [review]

**Pull request merged**